src/backend/src/round/round.service.ts

```diff
@@ -33,4 +35,19 @@ export class RoundService {
     await newRound.save();
     return newRound;
   }
+
+  async findByRoundNumber(
+    roundNumber: number,
+    gameId: number,
+  ): Promise<number> {
+    const game = await this.gameRepository.findOne(gameId, {
+      relations: ['rounds'],
+    });
+
+    const round = game.rounds.find((r) => r.roundNumber === roundNumber);
+    if (!round) {
+      throw new BadRequestException('Round number does not exist');
+    }
+    return round.id;
+  }
```

Comment:
Does it make sense for this to throw? Personally i don't expect find methods to throw. I would expect it to return the round || undefined.